contract/src/lib.rs

```diff
@@ -387,7 +387,8 @@ impl MpcContract {
                     // Make sure we have enough gas left to do 1 more call and clean up afterwards
                     // Observationally 30 calls < 300 TGas so 2 calls < 20 TGas
                     // We keep one call back so we can cleanup then call panic on the next call
-                    if depth > 29 {
+                    // Start cleaning up if there's less than 25 teragas left regardless of how deep you are.
+                    if depth > 30 || env::prepaid_gas() < Gas::from_tgas(25) {
```

Comment:
I feel like we should not have a set depth anymore, since this is purely already limited by the amount of gas a user specifies which is better for consistency. Users should feel free to burn however much gas they want to
```suggestion
                    if env::prepaid_gas() < Gas::from_tgas(25) {
```

Reply:
Right now we have a nice mapping between the timeout on the contract and the timeout on the MPC network. I'd probably need to rejig the congestion counter so that a bunch of requests didn't time out on the contract but not on the MPC service, or alternatively make the timeout on the MPC network configurable per request based on the gas attached.

Reply:
I'll look at that in a subsequent PR.